kimchi/src/circuits/expr.rs

```diff
@@ -1063,12 +1053,11 @@ impl<'a, F: FftField> EvalResult<'a, F> {
                     evals: es_sub,
                 },
             ) => {
-                let scale = (d_sub as usize) / (d as usize);
-                assert!(scale != 0);
-                evals.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
-                    *e += es_sub.evals[(scale * i + (d_sub as usize) * s) % es_sub.evals.len()];
-                });
-                Evals { evals, domain: d }
+                evals =
```

Comment:
This (and other similar changes) change the memory allocation characteristics (`par_iter` vs `par_iter_mut`). Perhaps adding helpers that cache all the intermediate terms in a struct (`ReducedDomainEvals`?) and use them to index / iterate over the structure would help?